### PR TITLE
[Backport v2.5] Better handling of still running etcd backups

### PR DESCRIPTION
**Pull request by jakefhyde**
Backport of https://github.com/rancher/rancher/pull/35132
Issue: https://github.com/rancher/rancher/issues/36038

Original PR description:

TLDR: Adds a check for backups that are currently running.

If a user sets the backup timeout to 10 minutes due to a slow S3-alike or similar, the backup will still be running during the next tick of the check interval. The cluster backup sync function will see that the last completed backup was longer than the backup interval (ignoring the one that's currently running) so it'll start a new backup and bulldoze the running one.

My addition basically checks that if since a backup's creation time is less than the backup config's timeout and "completed" is unknown. Then skip until the next iteration.

#34890

### CODE REVIEW

**Review by snasovich — APPROVED** [review]
The same changes as the source PR. Good to merge (only 1 "any" approval needed for straightforward backports)
